device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/actionsagent/actions/ActionManager.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iotsolutions.devicetelemetry.actionsagent.actions;
+
+import com.google.inject.Inject;
+import com.microsoft.azure.iotsolutions.devicetelemetry.actionsagent.models.AsaAlarmApiModel;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.exceptions.ResourceNotFoundException;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.models.actions.EmailAction;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.models.actions.IAction;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.runtime.IServicesConfig;
+import play.libs.ws.WSClient;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+public class ActionManager implements IActionManager {
+
+    private IActionExecutor actionExecutor;
+
+    @Inject
+    public ActionManager(IServicesConfig servicesConfig, WSClient wsClient) throws ResourceNotFoundException {
+        this.actionExecutor = new EmailActionExecutor(servicesConfig, wsClient);
+    }
+
+    public CompletionStage executeAsync(List<AsaAlarmApiModel> alarms) {
+        List<CompletionStage> tasks = new ArrayList<>();
+        for (AsaAlarmApiModel alarm : alarms) {
```

Comment:
Nit: You have an iterative style here and then a functional style right below it. Maybe choose one?